code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1030 @@
+			args: args{
+				urls: []string{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Consensus_ERR",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				consensus: 200,
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				consensus: 0,
+			},
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := MakeClientBalanceRequest(tt.args.clientID, tt.args.urls, tt.args.consensus)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MakeClientBalanceRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("MakeClientBalanceRequest() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetTransactionStatus(t *testing.T) {
+	txn := Transaction{
+		Hash:      encryption.Hash("data"),
+		Signature: "signature",
+	}
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				blob, err := json.Marshal(&txn)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				data := map[string]interface{}{
+					"txn": json.RawMessage(blob),
+				}
+				blob, err = json.Marshal(data)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	invServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				if _, err := rw.Write([]byte("}{")); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer invServer.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer errServer.Close()
+
+	nilTxnServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				txn := Transaction{
+					Hash: encryption.Hash("data"),
+				}
+				blob, err := json.Marshal(&txn)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				data := map[string]interface{}{
+					"txn": blob,
+				}
+				blob, err = json.Marshal(data)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer nilTxnServer.Close()
+
+	type args struct {
+		txnHash string
+		urls    []string
+		sf      int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *Transaction
+		wantErr bool
+	}{
+		{
+			name: "ERR",
+			args: args{
+				urls: []string{
+					"worng url",
+					errServer.URL,
+					invServer.URL,
+					nilTxnServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+			want:    &txn,
+			wantErr: false,
+		},
+		{
+			name:    "Txn_Not_Found_ERR",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Closure error, missing `tt:=tt`.

Reply:
There is no race between goroutines. Test running order is determined by tests struct slice.

Reply:
Fixed.